pkg/distributor/validate.go

```diff
@@ -194,13 +194,62 @@ type labelValueTooLongSummary struct {
 	sampleValue       string
 }
 
+type validationConfig struct {
+	samples  sampleValidationConfig
+	labels   labelValidationConfig
+	metadata metadataValidationConfig
+}
+
+// newValidationConfig builds a validationConfig based on the passed overrides.
+// TODO: This could still be more efficient, as each overrides call performs an atomic pointer retrieval and a lookup in a map,
```

Comment:
What are your thoughts on possibly making this more efficient in the future? Are you thinking about a low TTL cache to avoid the map calls? I was thinking about this but it could get tricky with invalidations if one of the fields changes.

Reply:
I was thinking about adding an `Overrides.Visit(userID string, func(limits *Limits))` that would retrieve the `*Limits` from runtime config once, and call the callback which would be used to fill our config.

However, some configs have some extra logic in `Overrides`, and letting this code access fields of `*Limits` seems like leaking responsibilities.

It would make sense to refactor the entire `Overrides` so methods would be on the `Limits` struct, not on the overrides itself, and Overrides would delegate the calls to `*Limits`, then we use that here.

Anyway, this is _good enough_, let's revisit in 2032.